### PR TITLE
refactor: more efficient and refactored queries

### PR DESCRIPTION
**Pull request by neilsong**
Feel free to add more optimizations and refactors.

### CODE REVIEW

**Comment by neilsong** [comment]
Currently old NetIDs without yalies data look like below. Should we filter or rename?

image

**Comment by Josh-Cena** [comment]
> Should we filter or rename?

I think it's fine

File: api/src/friends/friends.queries.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const existingRequest = await tx
+    .selectDistinctOn([
+      studentFriendRequests.netId,
+      studentFriendRequests.friendNetId,
+    ])
+    .from(studentFriendRequests)
+    .where(
+      and(
+        eq(studentFriendRequests.netId, friendNetId),
+        eq(studentFriendRequests.friendNetId, netId),
```
Comment (Josh-Cena) [inline api/src/friends/friends.queries.ts:21]: This is quite confusing. We should name the params as `from` and `to` for this function

**Comment by Josh-Cena** [comment]
@neilsong Please take another look; I've minimized the number of DB calls

File: api/src/friends/friends.handlers.ts [hunk 2]
```diff
@@ -217,39 +164,29 @@ export const requestAddFriend = async (
...
-          eq(studentFriendRequests.friendNetId, friendNetId),
+        or(
+          and(
+            eq(studentFriendRequests.netId, netId),
+            eq(studentFriendRequests.friendNetId, friendNetId),
+          ),
+          and(
+            eq(studentFriendRequests.netId, friendNetId),
+            eq(studentFriendRequests.friendNetId, netId),
+          ),
```
Comment (neilsong) [inline api/src/friends/friends.handlers.ts:182]: Is there ever a case where both exist? If this is the case, then the array destructuring won't work as intended

**Review by neilsong — COMMENTED** [review]
LGTM

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**